portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -21,23 +21,64 @@
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Iterator;
 import com.google.gson.Gson;
-
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> comments = new ArrayList<String>(
-      Arrays.asList("Wow such website", "Yeah, good. Okay.", "Chicken."));
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    String commentsJson = new Gson().toJson(comments);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    String queryString = request.getParameter("comment-capacity");
+    int commentCapacity = Integer.parseInt(queryString);
+
+    System.out.println("Preparing " + commentCapacity + " comments.");
+
+    List<String> commentsList = new ArrayList<>();
+    Iterator<Entity> entityIterator = results.asIterator();
+    for (int i = 0; i < commentCapacity; i++) {
+        
+        if (!entityIterator.hasNext()) break;
+        Entity entity = entityIterator.next();
+
+        long id = entity.getKey().getId();
+        String commentString = (String)entity.getProperty("commentString");
+        commentsList.add(commentString);
+    }
+
+    String commentsJson = new Gson().toJson(commentsList);
     response.setContentType("application/json;");
     response.getWriter().println(commentsJson);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String commentString = request.getParameter("comment-input");
+      if (commentString.length() > 0) {
```

Comment:
- Consider changing this to `!commentString.isEmpty()` which is a little more readable.
- If the commentString is empty, maybe print an error, or otherwise indicate that the comment wasn't saved because it was empty
- You can delete the extra line on line 71 :)